rttest/src/rttest.cpp

```diff
@@ -26,102 +26,55 @@
 
 #include <algorithm>
 #include <cassert>
-#include <cstdlib>
 #include <cmath>
 #include <fstream>
-#include <functional>
-#include <iostream>
+#include <ios>
 #include <map>
 #include <numeric>
+#include <ostream>
 #include <sstream>
+#include <stdexcept>
 #include <string>
 #include <utility>
 #include <vector>
 
-extern "C"
-{
 class rttest_sample_buffer
 {
 public:
   rttest_sample_buffer()
-  : latency_samples(nullptr),
-    major_pagefaults(nullptr),
-    minor_pagefaults(nullptr),
-    buffer_size(0)
-  {}
-  ~rttest_sample_buffer()
   {
-    resize(0);
   }
 
-  /// Free the current buffer and allocate a new buffer if new_buffer_size is greater than zero.
-  void resize(size_t new_buffer_size)
+  ~rttest_sample_buffer()
   {
-    if (new_buffer_size > 0) {
-      resize(0);
-      this->buffer_size = new_buffer_size;
-      this->latency_samples = static_cast<int64_t *>(
-        std::malloc(new_buffer_size * sizeof(int64_t)));
-      if (!this->latency_samples) {
-        fprintf(stderr, "Failed to allocate latency samples buffer\n");
-        exit(-1);
-      }
-      memset(this->latency_samples, 0, new_buffer_size * sizeof(int64_t));
-
-      this->major_pagefaults = static_cast<size_t *>(
-        std::malloc(new_buffer_size * sizeof(size_t)));
-      if (!this->major_pagefaults) {
-        fprintf(stderr, "Failed to allocate major pagefaults buffer\n");
-        exit(-1);
-      }
-      memset(this->major_pagefaults, 0, new_buffer_size * sizeof(size_t));
-
-      this->minor_pagefaults = static_cast<size_t *>(
-        std::malloc(new_buffer_size * sizeof(size_t)));
-      if (!this->minor_pagefaults) {
-        fprintf(stderr, "Failed to allocate minor pagefaults buffer\n");
-        exit(-1);
-      }
-      memset(this->minor_pagefaults, 0, new_buffer_size * sizeof(size_t));
-    } else {
-      if (this->latency_samples) {
-        free(this->latency_samples);
-        this->latency_samples = nullptr;
-      }
-      if (this->major_pagefaults) {
-        free(this->major_pagefaults);
-        this->major_pagefaults = nullptr;
-      }
-      if (this->minor_pagefaults) {
-        free(this->minor_pagefaults);
-        this->minor_pagefaults = nullptr;
-      }
-      this->buffer_size = 0;
-    }
   }
+
   rttest_sample_buffer(const rttest_sample_buffer & other)
   : rttest_sample_buffer()
   {
     *this = other;
   }
+
   void operator=(const rttest_sample_buffer & other)
```

Comment:
nit: the default `operator=` should be okey.

Reply:
Good call.  See e0124cf where I replaced all of this with the defaults.